### ISSUE TITLE
Window.clearcolor = None no longer working

### ISSUE THREAD

**Issue by Cheaterman** [kivy/kivy#8011]
**Software Versions**
* Python: Any
* OS: Any
* Kivy: 2.1.0
* Kivy installation method: Any

**Describe the bug**
Kivy 2.1.0 breaks Window.clearcolor = None - this issue was found in panda3d-kivy but might affect any library that relies on Kivy _not_ clearing the window (eg. in our case letting Panda3D do it).

See commit https://github.com/kivy/kivy/commit/6a20b0dd352fde77518062d7ba3c4bfb197d63cd - clearcolor=None used to be explicitly handled. I believe ColorProperty(allownone=True) might be an acceptable fix? If so, I could PR it.

**Expected behavior**
Kivy should not clear the window if Window.clearcolor is None ; instead, we get the following:

```
   File "/home/cheaterman/Dev/panda3d-kivy/env/lib/python3.10/site-packages/kivy/core/window/__init__.py", line 1531, in clear
     gl.glClearColor(*self.clearcolor)
 TypeError: kivy.graphics.opengl.glClearColor() argument after * must be an iterable, not NoneType
```

**To Reproduce**
Use panda3d-kivy v0.3.0 on Kivy v2.1.0 and try to run any of the examples - you will get a blackscreen.

**Code and Logs and screenshots**
https://i.imgur.com/29Qa2Wx.png

**Additional context**
See https://github.com/Cheaterman/panda3d-kivy/issues/22 for more info.

**Comment by pythonic64** [comment]
@Cheaterman I agree that this case is a regression. Can you make a pull request with your fix?
